Tests/PayID/PayIDUtilsTests.swift

```diff
@@ -2,63 +2,67 @@ import XCTest
 import XpringKit
 
 class PayIDUtilsTest: XCTestCase {
-  func testParsePaymentPointerHostAndPath() {
-    // GIVEN a payment pointer with a host and a path.
-    let rawPaymentPointer = "$example.com/foo"
+  func testParseValidPayID() {
+    // GIVEN a Pay ID with a host and a path.
+    let host = "xpring.money"
+    let path = "georgewashington"
+    let rawPayID = "\(path)$\(host)"
 
-    // WHEN it is parsed to a PaymentPointer object
-    let paymentPointer = PayIDUtils.parse(paymentPointer: rawPaymentPointer)
+    // WHEN it is parsed to components.
+    let payIDComponents = PayIDUtils.parse(payID: rawPayID)
 
     // THEN the host and path are set correctly.
-    XCTAssertEqual(paymentPointer?.host, "example.com")
-    XCTAssertEqual(paymentPointer?.path, "/foo")
+    XCTAssertEqual(payIDComponents?.host, host)
+    XCTAssertEqual(payIDComponents?.path, "/\(path)")
   }
 
-  func testParsePaymentPointerWithWellKnownPath() {
-    // GIVEN a payment pointer with a well known path.
-    let rawPaymentPointer = "$example.com"
+  func testParsePayIDTooManyDollarSigns() {
+    // GIVEN a Pay ID with too many '$'.
+    let host = "xpring$money" // Extra '$'
+    let path = "georgewashington"
+    let rawPayID = "\(path)$\(host)"
```

Comment:
I believe PayIDs can have as many `$`s as they want and still be a valid PayID (see https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-uri.md#definition).  

> Another scenario is a payment service provider (e.g., a digital wallet) that allows a user to use a PayID that is associated with some other payment service provider. For example, a user with the PayID "alice$bank.example.net" might register with a wallet website whose domain name is "wallet.example.net". In order to use her bank's PayID as the localpart of the wallet's 'payid' URI, the dollar-sign character (U+0024) needs to be percent-encoded as described in [RFC3986]. Thus, the resulting 'payid' URI would be "payid:alice%24bank.example$wallet.example".

Here, the PayID `payid:georgewashington$xpring$money` should be parsed to `payid:georgewashington%24xpring$money`.

Not suggesting you need to implement all of this, and likely no one will ask for something like that in the near future, but it's something to consider.